atomicfu/build.gradle.kts

```
@@ -0,0 +1,382 @@
+    tvosSimulatorArm64()
+    tvosX64()
+    tvosArm64()
+    iosArm64()
+
+    // Tier 3
+    androidNativeArm32()
+    androidNativeArm64()
+    androidNativeX86()
+    androidNativeX64()
+    mingwX64()
+    watchosDeviceArm64()
+
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    applyDefaultHierarchyTemplate {
+        group("native") {
+            group("nativeUnixLike") {
+                withLinux()
+                withApple()
+            }
+        }
+        group("androidNative32Bit") {
+            withAndroidNativeX86()
+            withCompilations { compilation ->
+                (compilation.target as? KotlinNativeTarget)?.konanTarget?.name == "android_arm32"
+            }
+        }
+        group("androidNative64Bit") {
+            withAndroidNativeArm64()
+            withAndroidNativeX64()
+        }
+
+    }
+
+    sourceSets {
+        val nativeUnixLikeMain by getting {
+            kotlin.srcDir("src/nativeUnixLikeMain/kotlin")
+            dependsOn(nativeMain.get())
+        }
+
+        val androidNative32BitMain by getting {
+            kotlin.srcDir("src/androidNative32BitMain/kotlin")
+            dependsOn(nativeMain.get())
+        }
+
+        val androidNative64BitMain by getting {
+            kotlin.srcDir("src/androidNative64BitMain/kotlin")
+            dependsOn(nativeMain.get())
+        }
+
+        val androidNative32BitTest by getting {
+            kotlin.srcDir("src/androidNative32BitTest/kotlin")
+            dependsOn(nativeTest.get())
+        }
+
+        val androidNative64BitTest by getting {
+            kotlin.srcDir("src/androidNative64BitTest/kotlin")
+            dependsOn(nativeTest.get())
+        }
+
+    }
+
+    // atomicfu-cinterop-interop.klib with an empty interop.def file will still be published for compatibility reasons (see KT-68411)
+    // This block can be removed when this issue in K/N compiler is resolved: KT-60874
+    targets.withType(KotlinNativeTarget::class).configureEach {
+        compilations.configureEach {
+            cinterops {
+                val interop by creating {
+                    definitionFile.set(project.file("src/nativeInterop/cinterop/interop.def"))
+                }
+            }
+        }
+    }
+}
+
+val transformer by configurations.creating
+
+dependencies {
+    transformer(project(":atomicfu-transformer"))
+}
+
+// ==== CONFIGURE JVM =====
+
+val classesPreAtomicFuDir = file("${layout.buildDirectory.get()}/classes/kotlin/jvm/test")
+val classesPostTransformFU = file("${layout.buildDirectory.get()}/classes/kotlin/jvm/postTransformedFU")
+val classesPostTransformVH = file("${layout.buildDirectory.get()}/classes/kotlin/jvm/postTransformedVH")
+val classesPostTransformBOTH = file("${layout.buildDirectory.get()}/classes/kotlin/jvm/postTransformedBOTH")
+
+tasks.withType(tasks.getByName("compileTestKotlinJvm").javaClass) {
+    (this as KotlinCompile).compilerOptions {
+        jvmTarget.set(JvmTarget.JVM_1_8)
+    }
+}
```

```suggestion
tasks.withType<KotlinCompile>().configureEach {
    compilerOptions {
        jvmTarget.set(JvmTarget.JVM_1_8)
    }
}
```

suggestion: a bit of tidying to make this snippet more Kotlin-esque.